src/protocol/attribution/apply_attribution_window.rs

```diff
@@ -20,31 +20,49 @@ use std::iter::{repeat, zip};
 ///
 /// # Errors
 /// Fails if sub-protocols fails.
-#[allow(dead_code)]
-async fn apply_attribution_window<F, C, T>(
+pub async fn apply_attribution_window<F, C, T>(
     ctx: C,
     input: &[MCApplyAttributionWindowInputRow<F, T>],
     attribution_window_seconds: u32,
-) -> Result<impl Iterator<Item = MCApplyAttributionWindowOutputRow<F, T>> + '_, Error>
+) -> Result<Vec<MCApplyAttributionWindowOutputRow<F, T>>, Error>
```

Comment:
Fine for now (temporary), but I think we will probably wind up converting everything back to iterators soon.